Source/Core/DolphinQt/Debugger/JITWidget.cpp

```diff
@@ -3,224 +3,654 @@
-  analyzer.SetDivByZeroExceptionsEnabled(Config::Get(Config::MAIN_DIVIDE_BY_ZERO_EXCEPTIONS));
-  analyzer.SetOption(PPCAnalyst::PPCAnalyzer::OPTION_CONDITIONAL_CONTINUE);
-  analyzer.SetOption(PPCAnalyst::PPCAnalyzer::OPTION_BRANCH_FOLLOW);
-
-  code_block.m_stats = &st;
-  code_block.m_gpa = &gpa;
-  code_block.m_fpa = &fpa;
-
-  if (analyzer.Analyze(ppc_addr, &code_block, &code_buffer, code_buffer.size()) != 0xFFFFFFFF)
-  {
-    std::string ppc_disasm_str;
-    auto ppc_disasm = std::back_inserter(ppc_disasm_str);
-    for (u32 i = 0; i < code_block.m_num_instructions; i++)
-    {
-      const PPCAnalyst::CodeOp& op = code_buffer[i];
-      const std::string opcode = Common::GekkoDisassembler::Disassemble(op.inst.hex, op.address);
-      fmt::format_to(ppc_disasm, "{:08x} {}\n", op.address, opcode);
-    }
+void JITWidget::OnPPCSymbolsUpdated()
+{
+  if (Core::GetState(m_system) != Core::State::Paused)
+    return;
+  CrossDisassemble();
+}
 
-    // Add stats to the end of the ppc box since it's generally the shortest.
-    fmt::format_to(ppc_disasm, "\n{} estimated cycles", st.numCycles);
-    fmt::format_to(ppc_disasm, "\nNum instr: PPC: {} Host: {}", code_block.m_num_instructions,
-                   host_instructions_disasm.instruction_count);
-    if (code_block.m_num_instructions != 0 && host_instructions_disasm.instruction_count != 0)
-    {
-      fmt::format_to(
-          ppc_disasm, " (blowup: {}%)",
-          100 * host_instructions_disasm.instruction_count / code_block.m_num_instructions - 100);
-    }
+void JITWidget::OnPPCBreakpointsChanged()
+{
+  // Whatever row(s) might have been selected could no longer exist, because adding or removing
+  // breakpoints can invalidate JIT blocks. We must clear the selection to avoid stale indices.
+  auto* const selection_model = m_table_view->selectionModel();
+  selection_model->clear();  // Side effect: currentChanged will be emitted (this is intended).
+  // Because currentChanged has been emitted, OnTableCurrentChanged has already handled the rest.
+}
 
-    fmt::format_to(ppc_disasm, "\nNum bytes: PPC: {} Host: {}", code_block.m_num_instructions * 4,
-                   host_instructions_disasm.code_size);
-    if (code_block.m_num_instructions != 0 && host_instructions_disasm.code_size != 0)
-    {
-      fmt::format_to(
-          ppc_disasm, " (blowup: {}%)",
-          100 * host_instructions_disasm.code_size / (4 * code_block.m_num_instructions) - 100);
-    }
+void JITWidget::OnConfigChanged()
+{
+  UpdateProfilingButton();
+}
 
-    m_ppc_asm_widget->setHtml(
-        QStringLiteral("<pre>%1</pre>").arg(QString::fromStdString(ppc_disasm_str)));
-  }
-  else
+void JITWidget::OnDebugFontChanged(const QFont& font)
+{
+  UpdateDebugFont(font);
+}
+
+void JITWidget::OnEmulationStateChanged(Core::State state)
+{
+  UpdateOtherButtons(state);
+  UpdateContent(state);
+}
+
+void JITWidget::OnRequestPPCComparison(u32 address, bool effective)
```

Comment:
The `effective` parameter name is a bit misleading. "Effective address" refers to either translating or not translating depending on what DR/IR is set to, but this variable controls translation directly, without deferring to DR/IR. `translate_address` would be a better parameter name.